ansible_runner/__main__.py

```diff
@@ -754,12 +765,15 @@ def main(sys_args=None):
     add_args_to_parser(isalive_container_group, DEFAULT_CLI_ARGS['container_group'])
     add_args_to_parser(transmit_container_group, DEFAULT_CLI_ARGS['container_group'])
 
-    if len(sys.argv) == 1:
-        parser.print_usage()
-        print_common_usage()
-        parser.exit(status=0)
+    try:
+        args = parser.parse_args(sys_args)
+    except SystemExit:
+        # If sys_args is None, assume this is interactive and use the behavior in the custom error() method.
+        if sys_args is None:
+            raise
 
-    args = parser.parse_args(sys_args)
+        # Otherwise, assume a call to main() passed in invalid arguments
+        parser.exit(2)
```

Comment:
wouldn't `raise` work in this case too? Invalid arguments are already processed correctly as it stands, which works as a `SystemExit` exception uncaught here from `parser.parse_args(sys_args)`

Reply:
Yeah, this whole thing is a bit convoluted. I don't love it.

The reason `raise` won't work is that the custom `error()` method changes the exit code. And, at least my from understanding of the existing code and tests, it should raise 2 when no args are passed in the Python interface but 0 when no args are passed in the CLI.

I'm really hoping someone will tell me "Oops we didn't mean to do that. It's fine to just exit with 2 in all cases."

```
> ansible-runner > /dev/null 2>&1 ; echo $status
0
> python -c 'from ansible_runner.__main__ import main; main([])' > /dev/null 2>&1 ; echo $status
2
```

If I change it to raise, the Python interface behavior changes.

```
> python -c 'from ansible_runner.__main__ import main; main([])' > /dev/null 2>&1 ; echo $status
0
```

Reply:
The 0 exit code looks like it was introduced in #333

I offer no defense, whatsoever, to giving an exit code of 0 for `ansible-runner` by itself. A code of 2 sounds _more_ correct. I care about that less than I care about making this code clean and maintainable.

Reply:
@matburt do you have any reason to keep the exit codes as they are?

Reply:
I got same feedback from @shanemcd and it seems we're ok with exiting 2 in all cases. I will work on simplifying this PR in a bit.

Reply:
The latest changes are _much_ cleaner.